applications/nrf5340_audio/src/bluetooth/bt_renderer/volume/bt_rend_vol.c

```diff
@@ -0,0 +1,370 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "bt_rend_vol_internal.h"
+
+#include <zephyr/kernel.h>
+#include <zephyr/types.h>
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/bluetooth/audio/vcp.h>
+
+#include "macros_common.h"
+#include "bt_rend.h"
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(bt_rend_vol, CONFIG_BT_REND_VOL_LOG_LEVEL);
+
+#define VOLUME_DEFAULT 195
+#define VOLUME_STEP    16
+
+static struct bt_vcp_vol_ctlr *vcs_client_peer[CONFIG_BT_MAX_CONN];
+
+/**
+ * @brief	Get the index of the first available vcs_client_peer
+ *
+ * @return	Index if success, -ENOMEM if no available indexes
+ */
+static int vcs_client_peer_index_free_get(void)
+{
+	for (int i = 0; i < ARRAY_SIZE(vcs_client_peer); i++) {
+		if (vcs_client_peer[i] == NULL) {
+			return i;
+		}
+	}
+
+	LOG_WRN("No more indexes for VCS peer clients");
+
+	return -ENOMEM;
+}
+
+/**
+ * @brief	Check if a given conn has a vcs_client_peer pointer
+ *
+ * @param	conn	The conn pointer to check
+ *
+ * @return	True if vcs client peer exists, False otherwise.
+ */
+static bool vcs_client_peer_exists(struct bt_conn *conn)
+{
+	int ret;
+
+	struct bt_conn *result_conn = NULL;
+
+	for (int i = 0; i < ARRAY_SIZE(vcs_client_peer); i++) {
+		ret = bt_vcp_vol_ctlr_conn_get(vcs_client_peer[i], &result_conn);
+
+		if (!ret && conn == result_conn) {
+			return true;
+		}
+
+		if (ret == -ENOTCONN) {
+			/* VCS client no longer connected, free the index */
+			vcs_client_peer[i] = NULL;
+			return false;
+		}
+	}
+
+	return false;
+}
+
+/**
+ * @brief  Callback handler for volume state changed.
+ *
+ * @note   This callback handler will be triggered if
+ *         volume state changed, or muted/unmuted.
+ */
+static void vcs_state_ctlr_cb_handler(struct bt_vcp_vol_ctlr *vcs, int err, uint8_t volume,
+				      uint8_t mute)
+{
+	int ret;
+
+	if (err) {
+		LOG_ERR("VCS state callback error: %d", err);
+		return;
+	}
+
+	for (int i = 0; i < ARRAY_SIZE(vcs_client_peer); i++) {
+		if (vcs == vcs_client_peer[i]) {
+			LOG_DBG("VCS state from remote device %d:", i);
+		} else {
+			LOG_DBG("Sync with other devices %d", i);
+			if (vcs_client_peer[i] != NULL) {
+				ret = bt_vcp_vol_ctlr_set_vol(vcs_client_peer[i], volume);
+				if (ret) {
+					LOG_DBG("Failed to sync volume to remote device %d, err = "
+						"%d",
+						i, ret);
+				}
+			}
+		}
+	}
+}
+
+/**
+ * @brief  Callback handler for VCS controller flags changed.
```

Comment:
```suggestion
 * @brief  Callback handler for the changed VCS controller flags.
```

Reply:
Will change to:
```suggestion
 * @brief  Callback handler for the VCS controller flags.
```
since it is written below in the note when this is triggered